backend/src/sms/services/sms-transactional.service.ts

```diff
@@ -0,0 +1,47 @@
+import { TemplateError } from 'postman-templating'
+import { SmsService, SmsTemplateService } from '@sms/services'
+import { TwilioCredentials } from '@sms/interfaces'
+import { loggerWithLabel } from '@core/logger'
+
+const logger = loggerWithLabel(module)
+
+/**
+ * Sanitizes an SMS message and sends it.
+ * @throws TemplateError if the body is invalid
+ * @throws Error if the message could not be sent
+ */
+async function sendMessage({
+  credentials,
+  body,
+  recipient,
+}: {
+  credentials: TwilioCredentials
+  body: string
+  recipient: string
+}): Promise<void> {
+  const sanitizedBody = SmsTemplateService.client.replaceNewLinesAndSanitize(
+    body
+  )
+  if (!sanitizedBody) {
+    throw new TemplateError(
+      'Message is invalid as it only contains invalid HTML tags.'
+    )
+  }
+
+  logger.info({
+    message: 'Sending transactional SMS',
+    action: 'sendMessage',
+  })
+  const sid = await SmsService.sendMessage(
+    credentials,
+    recipient,
+    sanitizedBody
+  )
+  if (!sid) {
```

Comment:
Will this line ever be reached, given that `twilioClient.send` already throws an error if `sid` is undefined.

Reply:
Ah, that's right. I'll do that.

Reply:
Resolved in https://github.com/opengovsg/postmangovsg/pull/1017/commits/689500ac36d16ac189ae5e2dc112ed3bf0d1143e